Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -14,6 +14,8 @@ internal protocol ExperienceRendering: AnyObject {
     func show(qualifiedExperiences: [Experience], priority: RenderPriority, completion: ((Result<Void, Error>) -> Void)?)
     func show(stepInCurrentExperience stepRef: StepReference, completion: (() -> Void)?)
     func dismissCurrentExperience(markComplete: Bool, completion: ((Result<Void, Error>) -> Void)?)
+    func getCurrentExperienceData() -> ExperienceData?
+    func getCurrentStepIndex() -> Experience.StepIndex?
```

Comment:
Just thinking back to previous POCs on embed experiences - I think this will still work great. It will just take the experience ID as a param and fetch the the appropriate state machine.

Reply:
Yeah, I had that in the back of my mind and it should work fine like you said 👌